File: python-sdk/examples/simple.py
"""Simple Python SDK Example

Usage:
    1. Start daemon: cargo run --package semantica-daemon
    2. Run example: python examples/simple.py
"""

import asyncio
from semantica import SematicaClient, EnqueueRequest


async def main():
    print("Semantica Python SDK - Simple Example")
    print("=" * 40)
    print()

    # 1. Connect to daemon
    print("1. Connecting to daemon...")
    async with SematicaClient("http://127.0.0.1:9527") as client:
        print("   ✓ Connected\n")

        # 2. Enqueue a job
        print("2. Enqueuing a job...")
        response = await client.enqueue(
            EnqueueRequest(
                job_type="INDEX_FILE",
                queue="default",
                subject_key="examples/simple.py",
                payload={"path": "examples/simple.py", "mode": "full_index"},
                priority=5,
            )
        )

        print("   ✓ Job enqueued:")
        print(f"     - ID: {response.job_id}")
        print(f"     - State: {response.state}")
        print(f"     - Queue: {response.queue}\n")

        # 3. Wait a bit
        print("3. Waiting 2 seconds...")
        await asyncio.sleep(2)
        print("   ✓ Done\n")

        # 4. Tail logs
        print("4. Fetching job logs...")
        logs_response = await client.tail_logs(response.job_id, lines=10)

        print("   ✓ Logs retrieved:")
        if logs_response.log_path:
            print(f"     - Path: {logs_response.log_path}")
        print(f"     - Lines: {len(logs_response.lines)}")

        if logs_response.lines:
            print(f"\n   Last {len(logs_response.lines)} lines:")
            for line in logs_response.lines:
                print(f"     | {line}")
        print()

        # 5. Cancel job
        print("5. Cancelling job...")
        cancel_response = await client.cancel(response.job_id)

        if cancel_response.cancelled:
            print("   ✓ Job cancelled")
        else:
            print("   ⚠ Job was already finished")

        print("\n✓ Example completed successfully!")


if __name__ == "__main__":
    asyncio.run(main())

File: crates/sdk/examples/simple.rs
//! Simple SDK Example
//!
//! Demonstrates basic usage of the Semantica SDK.
//!
//! # Usage
//!
//! 1. Start the daemon:
//!    ```bash
//!    cargo run --package semantica-daemon
//!    ```
//!
//! 2. Run this example:
//!    ```bash
//!    cargo run --example simple
//!    ```

use semantica_sdk::{EnqueueRequest, SematicaClient};
use serde_json::json;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("Semantica SDK - Simple Example");
    println!("================================\n");

    // 1. Connect to daemon
    println!("1. Connecting to daemon...");
    let client = SematicaClient::connect("http://127.0.0.1:9527").await?;
    println!("   ✓ Connected\n");

    // 2. Enqueue a job
    println!("2. Enqueuing a job...");
    let enqueue_response = client
        .enqueue(EnqueueRequest {
            job_type: "INDEX_FILE".to_string(),
            queue: "default".to_string(),
            subject_key: "examples/simple.rs".to_string(),
            priority: 5,
            payload: json!({
                "path": "examples/simple.rs",
                "mode": "full_index"
            }),
        })
        .await?;

    println!("   ✓ Job enqueued:");
    println!("     - ID: {}", enqueue_response.job_id);
    println!("     - State: {}", enqueue_response.state);
    println!("     - Queue: {}\n", enqueue_response.queue);

    // 3. Wait a bit for processing
    println!("3. Waiting 2 seconds...");
    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
    println!("   ✓ Done\n");

    // 4. Tail logs
    println!("4. Fetching job logs...");
    let logs_response = client.tail_logs(&enqueue_response.job_id, Some(10)).await?;

    println!("   ✓ Logs retrieved:");
    if let Some(log_path) = &logs_response.log_path {
        println!("     - Path: {}", log_path);
    }
    println!("     - Lines: {}", logs_response.lines.len());

    if !logs_response.lines.is_empty() {
        println!("\n   Last {} lines:", logs_response.lines.len());
        for line in &logs_response.lines {
            println!("     | {}", line);
        }
    }
    println!();

    // 5. Cancel the job (if still running)
    println!("5. Cancelling job...");
    let cancel_response = client.cancel(&enqueue_response.job_id).await?;

    if cancel_response.cancelled {
        println!("   ✓ Job cancelled");
    } else {
        println!("   ⚠ Job was already finished");
    }

    println!("\n✓ Example completed successfully!");

    Ok(())
}
